src/extractor/scripting_environment_lua.cpp

```diff
@@ -525,8 +526,16 @@ void Sol2ScriptingEnvironment::ProcessElements(
                     resulting_ways.push_back(std::make_pair(x, std::move(result_way)));
                     break;
                 case osmium::item_type::relation:
-                    resulting_restrictions.push_back(restriction_parser.TryParse(
-                        static_cast<const osmium::Relation &>(*entity)));
+                    result_res.clear();
+                    result_res =
+                        restriction_parser.TryParse(static_cast<const osmium::Relation &>(*entity));
+                    if (result_res.size() > 0)
```

Comment:
Size check is not necessary, the for loop would just not do any iterations in that case.